packages/runtime/src/runtimes/react/controls/style-v2.ts

```diff
@@ -0,0 +1,69 @@
+import { useId } from 'react'
+import {
+  ControlDefinition,
+  StyleV2ControlData,
+  StyleV2ControlDefinition,
+  StyleV2Format,
+} from '../../../controls'
+import { useStyle } from '../use-style'
+import {
+  findBreakpointOverride,
+  getBaseBreakpoint,
+  getBreakpointMediaQuery,
+} from '../../../state/modules/breakpoints'
+import { useBreakpoints } from '..'
+import { CSSObject } from '@emotion/css'
+
+function useStyleControlCssObject(
+  styleControlData: StyleV2ControlData,
+  controlDefinition: StyleV2ControlDefinition,
+): CSSObject {
+  const breakpoints = useBreakpoints()
+
+  return {
+    ...controlDefinition.config.getStyle(
+      findBreakpointOverride(breakpoints, styleControlData, getBaseBreakpoint(breakpoints).id)
+        ?.value,
+    ),
+    ...breakpoints.reduce((styles, breakpoint) => {
+      return {
+        ...styles,
+        [getBreakpointMediaQuery(breakpoint)]: controlDefinition.config.getStyle(
+          findBreakpointOverride(breakpoints, styleControlData, breakpoint.id)?.value,
+        ),
+      }
+    }, {}),
+  }
+}
+
+export type StyleV2ControlFormattedValue = CSSObject | string
+
+export type ResolveStyleV2ControlFormattedValue<T extends ControlDefinition> =
+  T extends StyleV2ControlDefinition
+    ? undefined extends T['config']['format']
+      ? string | undefined
+      : T['config']['format'] extends typeof StyleV2Format.Class
+      ? string | undefined
+      : T['config']['format'] extends typeof StyleV2Format.Object
+      ? CSSObject | undefined
+      : never
+    : never
+
+export function useFormattedStyleV2<T extends StyleV2ControlDefinition>(
+  styleControlData: StyleV2ControlData,
+  controlDefinition: T,
+): StyleV2ControlFormattedValue {
```

Comment:
When I try to use `ResolveStyleV2ControlFormattedValue` all the returns in the function as "Unassignable to the generic type"

Reply:
I don't understand what you're trying to say here

Reply:
I can ignore this for now since I have to cast the value anyways.

--- 


Further explanation: 

Similar to the Image control I have two types. The one that is prefixed with `Resolve` is generic and should be useable with `useFormattedStyleV2`, but I am struggling to do so. 


<img width="980" alt="CleanShot 2023-05-24 at 10 15 21@2x" src="https://github.com/makeswift/makeswift/assets/20950876/dd4f5219-14c6-4529-86dc-d46bda576f33">

Reply:
This type was ultimately removed so I am resolving this conversation